src/config.rs

```diff
@@ -14,15 +14,15 @@ use url2::Url2;
 use crate::{HcHttpGatewayError, HcHttpGatewayResult};
 
 /// Default payload size limit (10 kilobytes)
-pub const DEFAULT_PAYLOAD_LIMIT_BYTES: u32 = 10 * 1024;
+pub const DEFAULT_PAYLOAD_LIMIT_BYTES: usize = 10 * 1024;
```

Comment:
Changed the type to `usize` for platform compatibility

Reply:
> Does it need to ever be a u64?

Not really, using usize seems like the more the idiomatic Rust choice for memory sizes and counts as it automatically adjusts to the platform

Reply:
It's more important that the code accurately expresses intent. If it's never valid for this to be larger than `u32::MAX` then I think we're losing clarity by choosing `usize`

Reply:
I see, reverted to `u32`